auth/opa/rpcauth/input.go

```diff
@@ -45,11 +45,14 @@ type RPCAuthInput struct {
 	// Raw grpc metdata associated with this call.
 	Metadata metadata.MD `json:"metadata"`
 
-	// Information about the calling peer, if available
+	// Information about the calling peer, if available.
 	Peer *PeerAuthInput `json:"peer"`
 
 	// Information about the host serving the RPC.
 	Host *HostAuthInput `json:"host"`
+
+	// Implementation specific extensions.
+	Extensions []interface{} `json:"extensions"`
```

Comment:
as mentioned in chat:

I think this might be nicer as json.RawMessage, rather than []interface

reasoning:
 - it makes it clear that the value needs to be convertible to json (and sidesteps ensuring that it's cleanly convertible, and doesn't have some custom marshaller that runs for infinite time or panics)
 - it's "cheaper", in the case that this is just some static list of attributes (imagine: additional host attributes/tags/etc) which only need to be marshalled once, and then just stuffed in by a hook (MarshalJSON for RawMessage is a noop)

Reply:
Yep, as discussed cleaner API for end users so converted over.